packages/state-transition/src/block/processBlsToExecutionChange.ts

```diff
@@ -23,39 +28,46 @@ export function processBlsToExecutionChange(
   validator.withdrawalCredentials = newWithdrawalCredentials;
 }
 
-export function assertValidBlsToExecutionChange(
+export function isValidBlsToExecutionChange(
   state: CachedBeaconStateCapella,
   signedBLSToExecutionChange: capella.SignedBLSToExecutionChange,
   verifySignature = true
-): void {
+): {valid: boolean; error?: Error} {
   const addressChange = signedBLSToExecutionChange.message;
 
   if (addressChange.validatorIndex >= state.validators.length) {
-    throw Error(
+    const error = Error(
       `withdrawalValidatorIndex ${addressChange.validatorIndex} > state.validators len ${state.validators.length}`
     );
+    return {valid: false, error};
```

Comment:
Just do everywhere, no need to declare error variable
```suggestion
    return {valid: false, error: Error(...)};
```